feat: add companies to resource page

bottbott opened the pull request:
## What issue is this referencing?
closes #364

Here's what things look like after the change.
Top
<img width="1019" alt="image" src="https://github.com/user-attachments/assets/db3592ae-de8f-4059-b0ab-36e745655363">
Bottom
<img width="1034" alt="image" src="https://github.com/user-attachments/assets/49a3cb69-04cb-4913-accf-c1de656a43c7">


## Do these code changes work locally and have you tested that they fix the issue yourself?

-   [x] yes!

## Does the following command run without warnings or errors?

-   [ ] yes! `npm run pr-checks`
There's an existing error in there about an unused variable. Left it since the code is out of scope. Let me know if anyone wants it fixed - happy to do it if we want it to go away in this PR.

## Have you taken a look at our [contributing guidelines](https://github.com/devedmonton/DES-Website/blob/main/.github/CONTRIBUTING.md)?

-   [x] yes!

## My node version matches the one suggested when running `nvm use`?

-   [x] yes!

MandyMeindersma commented on constants/companies.ts:
```diff
@@ -0,0 +1,202 @@
+  },
+  {
+    name: 'Scopear',
+    to: 'https://www.scopear.com/jobs/',
```
nitpick, I keep reading "ear" here hahahah Can we capitalize the AR?

MandyMeindersma commented on constants/companies.ts:
```diff
@@ -0,0 +1,202 @@
+    name: 'Paper Leaf',
+    to: 'https://paper-leaf.com/careers/',
+    description: 'Paper Leaf is a web development and design agency, creating custom web applications for businesses. Developers at Paper Leaf work on frontend and backend solutions, building responsive and user-friendly websites and apps for a variety of industries.',
+  },
```
One I added late last week is ualberta too, if you want to add it that would be great. Otherwise I can create another issue to add it. 

https://apps.ualberta.ca/careers/list/category/50

MandyMeindersma commented:
Wow! You added all the descriptions too! This is really impressive work! Thank you so much I am so excited to have this on the site!

I am thinking I maybe want to move the contacts in the header to the footer and then we can highlight "companies" in the header but I think that will be another issue! I don't want to hold up your work! :)

bottbott commented:
fyi @MandyMeindersma - i added ualberta and fixed that ScopeAR typo. 

Let me know if you decide that it would be preferable to shuffle the header/footer around to make this more prominent.

MandyMeindersma commented:
Great work @bottbott

MandyMeindersma commented:
@allcontributors add @bottbott for code and documentation

MandyMeindersma commented:
and here is the new issue to highlight your work more! Thank you again!
